src/implementations/twig/components/banner/banner.html.twig

```diff
@@ -68,19 +124,72 @@
 {# Print the result #}
 
 <section class="{{ _css_class }}"{{ _extra_attributes|raw }}>
-{% if _picture.img is not empty and _variant != 'plain-background'  %}
-  <figure class="ecl-banner__picture-container">
-    {% include '@ecl/picture/picture.html.twig' with {
-      picture: _picture,
-      extra_classes: 'ecl-banner__picture',
-      extra_image_classes: 'ecl-banner__image',
-      extra_attributes: [{ name: 'data-ecl-banner-image' }],
-    } only %}
-  {% if _credit is not empty and _variant != 'plain-background' %}
-    <footer class="ecl-banner__credit"><small>{{ _credit }}</small></footer>
+  {# Media #}
+  {% if _picture.img is not empty %}
+    {# Image #}
+    <figure class="ecl-banner__picture-container">
+      {% include '@ecl/picture/picture.html.twig' with {
+        picture: _picture,
+        extra_classes: 'ecl-banner__picture',
+        extra_image_classes: 'ecl-banner__image',
+        extra_attributes: [{ name: 'data-ecl-banner-image' }],
+      } only %}
+    {% if _credit is not empty %}
+      <footer class="ecl-banner__credit"><small>{{ _credit }}</small></footer>
+    {% endif %}
+    </figure>
+  {% elseif _video is not empty %}
+    {# Video #}
+    {% set _video_sources = _video.sources|default({}) %}
+    {% set _video_tracks = _video.tracks|default({}) %}
+    <div class="ecl-banner__video-container">
+      <video class="ecl-banner__video" poster="{{ _video.poster }}" autoplay loop muted data-ecl-banner-video>
+        {% if _video_sources is not empty and _video_sources is iterable %}
+          {% for video in _video_sources %}
+            <source src="{{ video.src }}" type="{{ video.type }}">
+          {% endfor %}
+        {% endif %}
+        {% if _video_tracks is not empty and _video_tracks is iterable %}
+          {% for track in _video_tracks %}
+            <track src="{{ track.src }}" kind="{{ track.kind }}" srclang="{{ track.src_lang }}" label="{{ track.label }}">
```

Comment:
we never had issues with it in the media container but this lacks a number of checks, all these attributes shoult not be set if there is no value for them. If we move this into a dedicated template we might take care of this there.